routing/endpointregistry_test.go

```diff
@@ -50,20 +51,34 @@ func TestSetAndGetAnotherKey(t *testing.T) {
 }
 
 func TestDoRemovesOldEntries(t *testing.T) {
-	route := &Route{LBEndpoints: []LBEndpoint{
-		{Host: "existing endpoint"},
+	r := routing.NewEndpointRegistry(routing.RegistryOptions{})
+
+	route := &routing.Route{LBEndpoints: []routing.LBEndpoint{
+		{Host: "endpoint1.test:80"},
+		{Host: "endpoint2.test:80"},
 	}}
 	route.BackendType = eskip.LBBackend
+	r.Do([]*routing.Route{route})
+
+	mExist := r.GetMetrics("endpoint1.test:80")
+	mExistYet := r.GetMetrics("endpoint2.test:80")
+	assert.NotEqual(t, time.Time{}, mExist.DetectedTime())
+	assert.NotEqual(t, time.Time{}, mExistYet.DetectedTime())
 
-	r := NewEndpointRegistry(RegistryOptions{})
+	r.IncInflightRequest("endpoint1.test:80")
+	r.IncInflightRequest("endpoint2.test:80")
 
-	r.IncInflightRequest("existing endpoint")
-	r.IncInflightRequest("removed endpoint")
-	r.lastSeen["removed endpoint"] = time.Now().Add(-lastSeenTimeout)
-	r.Do([]*Route{route})
+	routing.SetNow(r, func() time.Time {
+		return time.Now().Add(routing.ExportLastSeenTimeout + time.Second)
+	})
+	route = &routing.Route{LBEndpoints: []routing.LBEndpoint{
+		{Host: "endpoint1.test:80"},
+	}}
+	route.BackendType = eskip.LBBackend
```

Comment:
This could be a part of struct initializer

Reply:
That's leftover from the time where this code was part of `routing` package. Will fix.